dashboard/app/models/authentication_option.rb

```diff
@@ -26,12 +26,17 @@ class AuthenticationOption < ApplicationRecord
   belongs_to :user
 
   # These are duplicated from the user model, until we're ready to cut over and remove them from there
-  before_save :normalize_email, :hash_email, :fix_by_user_type
+  before_save :normalize_email, :hash_email, :remove_student_cleartext_email,
+    :fill_authentication_id
 
-  def fix_by_user_type
+  def remove_student_cleartext_email
     self.email = '' if user.student?
   end
 
+  def fill_authentication_id
+    self.authentication_id = hashed_email if credential_type == 'email'
+  end
```

Comment:
Moved this into a hook because it's appropriate to do even when we don't need to `hash_email` because we have no email and the hashed email is already present.